### PR TITLE
fix(quantic): fixed quantic RecentQueries onClick issue

### PR DESCRIPTION
**Pull request by pkuttappan**
[SFINT-4040](https://coveord.atlassian.net/browse/SFINT-4080) Fixed issue related to the query not being triggered on being clicked

https://user-images.githubusercontent.com/86019421/132927005-f8f72061-403d-4410-a769-03c4b14ad47e.mov

### CODE REVIEW

**Review by jeremierobert-coveo — COMMENTED** [review]
Do you think we can have a test for this behavior ?

**Comment by pkuttappan** [comment]
> Do you think we can have a test for this behavior ?

yes i think, we could check for the query param being set in the url to test this behaviour. Could have a test like that.

**Comment by nathanlb** [comment]
> > Do you think we can have a test for this behavior ?
> 
> yes i think, we could check for the query param being set in the url to test this behaviour. Could have a test like that.

Sounds like a great time to add a cypress tests for recent queries! 🎉

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html [hunk 1]
```diff
@@ -14,8 +14,8 @@
         <div class="slds-text-body_small empty-list-message slds-p-horizontal_medium slds-var-m-vertical_small">{labels.emptyListLabel}</div>
       </template>
       <ul>
-        <template for:each={queries} for:item="query">
-          <li class="slds-var-m-vertical_x-small slds-grid" key={query}>
+        <template for:each={queries} for:item="query" for:index="index">
+          <li key={query} value={index} onclick={executeQuery} class="slds-var-m-vertical_x-small slds-grid">
             <div class="slds-col recent-query__container">
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html:19]: Just noticing that now, but I believe it would be a better idea to use the anchor html element for those I think:
https://www.lightningdesignsystem.com/utilities/interactions/#Link-Reset-with-Text-Link-Inside
You should inherit already from the hover styles and branding from the community most likely.

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css [hunk 2]
```diff
@@ -10,15 +10,20 @@
...
 li:hover {
   text-decoration: underline;
   cursor: pointer;
-}
\ No newline at end of file
+  text-decoration-color: var(--brand-text-link, #0176d3);
+}
+
+li div {
+  pointer-events: none;
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css:28]: Why is this needed?

Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css:28]: IIRC without, this clicks on the icon or actual label would not cause the search to execute. (seems like it might relate to the issue at hand). It's possible it was a suggestion I made prior to some html changes that might cause it not to work anymore.

Comment (pkuttappan) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css:28]: so that the onClick of the text and the icon takes the onClick of the parent . In other words to avoid the onclicks of the child (divs)

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html (hunk 1 above)
Comment (pkuttappan) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html:19]: On using the anchor element, there are a few issues I am facing. The first one is the value shows as undefined in the executeQuery for which I did get a solution, but the solution disables the pointer and the underline on hover. So i have to again add li:hover in css


`<li key={query} value={index} onclick={executeQuery} class="slds-var-m-vertical_x-small slds-grid">

            <a href="#" class="slds-col recent-query__container">

              <lightning-icon class="recent-query__icon slds-current-color" icon-name="utility:search" alternative-text="search icon" size="xx-small"></lightning-icon>

              <span class="slds-text-body_small nowrap query-text__container slds-truncate slds-text-link">
               {query}
              </span>

            </a>

          </li>
`
the css includes:
 li a {pointer-events:none;}

Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html:19]: mmm yes i agree. would be easier to keep it consistent with the recent-results-list which uses `<a>` tags as well

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css [hunk 3]
```diff
@@ -2,23 +2,17 @@
...
 }
 
 .empty-list-message {
-  color: var(--sds-c-color-gray-8, #8E959D);
-}
-
-li:hover {
-  text-decoration: underline;
-  cursor: pointer;
+  color: var(--sds-c-color-gray-8, #8e959d);
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css:17]: This `--sds-c-color-gray-8` design token actually [isn't available](https://www.lightningdesignsystem.com/design-tokens/#category-color) to devs so apparently it'll always use the backup.

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html [hunk 4]
```diff
@@ -14,14 +14,14 @@
...
-            <div class="slds-col recent-query__container">
+        <template for:each={queries} for:item="query" for:index="index">
+          <li key={query} class="slds-var-m-vertical_x-small slds-grid" value={index} onclick={executeQuery}>
+            <a class="slds-col recent-query__container">
               <lightning-icon class="recent-query__icon slds-current-color" icon-name="utility:search" alternative-text="search icon" size="xx-small"></lightning-icon>
-              <div class="slds-text-body_small nowrap query-text__container">
-                <p class="slds-truncate">{query}</p>
-              </div>
-            </div>
+              <span class="slds-text-body_small nowrap query-text__container slds-truncate slds-text-link">
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html:21]: Have you tested that the `slds-truncate` works well on this span element as well?

**Review by nathanlb — APPROVED** [review]

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html (hunk 4 above)
Comment (pkuttappan) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html:21]: yes

Comment (nathanlb) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html:21]: nice 👍

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css (hunk 3 above)
Comment (pkuttappan) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css]: ohh! i"ll change it to just the color code then

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html (hunk 1 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.html]: Yea, I believe this is a problem `li a {pointer-events:none;}` and that you should override it by adding a class into it.

`li a.my-class {pointer-events:auto;}`

**Review by erocheleau — APPROVED** [review]
As long as it's a <a> element now it's perfect 👍

File: packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css (hunk 2 above)
Comment (pkuttappan) [reply packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/quanticRecentQueriesList.css]: previously I was using event.target, for which the events weren't bubbling up but now on using event.currentTarget, the issue gets solved. I don't need to use pointer-events: none for child divs
